### ISSUE TITLE
Experiment tracking integration (Weights & Biases / MLflow)

### ISSUE THREAD

**Issue by NP-compete** [NP-compete/smithery#40]
## Problem

Training without experiment tracking is flying blind. Users need to compare runs, track hyperparameters, visualize loss curves, and share results. The \`[wandb]\` extra exists in pyproject.toml but there's no integration code.

### W&B Integration (Primary)

Auto-detect and configure W&B when the \`wandb\` package is installed:

\`\`\`python
# In AgentTrainer.train():
if wandb_available():
    wandb.init(
        project="smithery",
        config={
            "base_model": config.base_model,
            "lora_r": config.lora.r,
            "lora_alpha": config.lora.alpha,
            "epochs": config.training.epochs,
            "learning_rate": config.training.learning_rate,
            "num_tools": len(tools),
            "num_examples": len(dataset),
        },
        tags=["smithery", config.base_model.split("/")[-1]],
    )
\`\`\`

### What to Log

| Event | What to Log | When |
|---|---|---|
| Training start | Config, tool definitions, data stats | Once |
| Each step | Loss, learning rate, GPU memory | Every step |
| Each epoch | Eval loss, sample tool calls, cost estimate | End of epoch |
| Training end | Final metrics, adapter size, total cost | Once |
| Evaluation | All EvalMetrics, per-tool breakdown | After eval |

### Sample Tool Call Logging

During training, periodically generate sample tool calls and log them as W&B Tables:

| Epoch | Prompt | Expected Tool | Predicted Tool | Correct? |
|---|---|---|---|---|
| 1 | "Weather in Tokyo?" | get_weather | get_forecast | No |
| 2 | "Weather in Tokyo?" | get_weather | get_weather | Yes |
| 3 | "Weather in Tokyo?" | get_weather | get_weather | Yes |

### MLflow Support (Secondary)

For users who prefer MLflow (common in enterprise):

\`\`\`python
smithery train --config ... --data ... --tracker mlflow --tracking-uri http://localhost:5000
\`\`\`

### CLI Flags

\`\`\`bash
smithery train --config ... --data ... \
  --tracker wandb \                    # or mlflow, or none
  --wandb-project my-project \         # optional override
  --wandb-entity my-team               # optional
\`\`\`

### No Tracking by Default

If neither wandb nor mlflow is installed, training works fine with console output only. No errors, no warnings. Tracking is opt-in.

## Acceptance Criteria

- [ ] W&B integration logs config, loss, LR, memory, sample tool calls, eval metrics
- [ ] MLflow integration logs the same data
- [ ] Auto-detect: if wandb installed, offer to use it; if not, skip silently
- [ ] \`--tracker\` CLI flag to choose (wandb / mlflow / none)
- [ ] Sample tool call table logged at each epoch
- [ ] No dependency on wandb/mlflow unless user installs the extra
- [ ] Unit tests with mocked tracking calls

## Dependencies

- Depends on #8 (training pipeline)
- \`wandb\` is already in optional dependencies

## Estimated Effort

Medium (3-5 days). W&B and MLflow both have well-documented Python APIs. The sample tool call logging requires inference during training, which needs careful memory management.
